pallets/automation-time/src/lib.rs

```diff
@@ -1129,12 +1350,43 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Reschedules an existing task for a given number of execution times
+		fn reschedule_existing_task(
+			task_id: T::Hash,
+			who: T::AccountId,
+			action: &Action<T>,
+			execution_times: Vec<UnixTime>,
+		) -> Result<(), DispatchError> {
+			let new_executions = execution_times.len().try_into().unwrap();
+			let fee = Self::calculate_execution_fee(action, new_executions)?;
+			T::NativeTokenExchange::can_pay_fee(&who, fee.clone())
+				.map_err(|_| Error::<T>::InsufficientBalance)?;
+
+			Self::insert_scheduled_tasks(task_id, execution_times.clone())?;
+
+			T::NativeTokenExchange::withdraw_fee(&who, fee.clone())
+				.map_err(|_| Error::<T>::LiquidityRestrictions)?;
+
+			Self::deposit_event(Event::<T>::TaskScheduled { who, task_id });
+			Ok(())
+		}
+
 		pub fn generate_task_id(owner_id: AccountOf<T>, provided_id: Vec<u8>) -> T::Hash {
 			let task_hash_input =
 				TaskHashInput::<T> { owner_id: owner_id.clone(), provided_id: provided_id.clone() };
 			T::Hashing::hash_of(&task_hash_input)
 		}
 
+		pub fn generate_auto_compound_delegated_stake_provided_id(
+			delegator: &AccountOf<T>,
+			collator: &AccountOf<T>,
+		) -> Vec<u8> {
+			let mut provided_id = "AutoCompoundDelegatedStake".as_bytes().to_vec();
+			provided_id.extend(delegator.encode());
+			provided_id.extend(collator.encode());
```

Comment:
I had a lot of trouble getting the account ids to convert into something that I could use for a provided id and ended up here. It's not clear to me what the `encode` method is doing other than that it converts the account id to a byte string. I expected the resulting hex to be the public key of the account but it doesn't seem to be. Let me know if any of you know any less opaque conversions.